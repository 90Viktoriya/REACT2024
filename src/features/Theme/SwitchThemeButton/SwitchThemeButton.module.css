.switch {
  cursor: pointer;
  background-color: var(--brown);
  border: 2px outset var(--dark);
  color: var(--purple);
  border-radius: 8px;
  width: 100px;
  height: 40px;
  margin-left: 2em;
}