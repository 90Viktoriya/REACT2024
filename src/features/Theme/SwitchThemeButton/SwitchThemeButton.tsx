import { useCallback } from 'react';
import { ComponentsCaptions } from '../../../data/ComponentsCaptions';
import { useTheme } from '../../../hooks/useTheme';
import styles from './SwitchThemeButton.module.css';

export function SwitchThemeButton() {
  const { toggleTheme } = useTheme();

  const handleOnClick = useCallback(() => {
    toggleTheme();
  }, [toggleTheme]);

  return (
    <button className={styles.switch} onClick={handleOnClick}>
      {ComponentsCaptions.SWITCH_THEME}
    </button>
  );
}
