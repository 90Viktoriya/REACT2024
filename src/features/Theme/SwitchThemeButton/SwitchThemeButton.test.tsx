import { describe, expect } from 'vitest';
import { render } from '@testing-library/react';
import { SwitchThemeButton } from './SwitchThemeButton';

describe('Given SwitchThemeButton component', () => {
  it('when rendered, should match snapshot', () => {
    const { asFragment } = render(<SwitchThemeButton />);

    expect(asFragment()).toMatchSnapshot();
  });
});
