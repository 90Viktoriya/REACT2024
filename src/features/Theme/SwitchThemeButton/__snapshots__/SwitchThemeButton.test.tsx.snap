// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given SwitchThemeButton component > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="_switch_bea301"
  >
    SWITCH theme
  </button>
</DocumentFragment>
`;
